aries_cloudagent/indy/sdk/profile.py

```diff
@@ -87,8 +88,59 @@ def bind_providers(self):
                 "aries_cloudagent.storage.vc_holder.indy.IndySdkVCHolder", self.opened
             ),
         )
-
-        if self.ledger_pool:
+        if (
+            self.settings.get("ledger.ledger_config_list")
+            and len(self.settings.get("ledger.ledger_config_list")) >= 1
+        ):
+            write_ledger_config = None
+            prod_write_ledger_pool = []
+            non_prod_write_ledger_pool = []
+            for ledger_config in self.settings.get("ledger.ledger_config_list"):
+                if ledger_config.get("is_production") and ledger_config.get("is_write"):
+                    prod_write_ledger_pool.append(ledger_config)
+                elif not ledger_config.get("is_production") and ledger_config.get(
+                    "is_write"
+                ):
+                    non_prod_write_ledger_pool.append(ledger_config)
+            cache = self.context.injector.inject_or(BaseCache)
+            if len(prod_write_ledger_pool) >= 1:
+                write_ledger_config = prod_write_ledger_pool[0]
+            elif len(non_prod_write_ledger_pool) >= 1:
+                write_ledger_config = non_prod_write_ledger_pool[0]
+            else:
+                raise ProfileError(
+                    "No write ledger configuration found in ledger_config_list which "
+                    "was provided with --genesis-transactions-list"
+                )
+            injector.bind_provider(
+                BaseLedger,
+                ClassProvider(
+                    IndySdkLedger,
+                    IndySdkLedgerPool(
+                        write_ledger_config.get("pool_name")
+                        or write_ledger_config.get("id"),
+                        keepalive=write_ledger_config.get("keepalive"),
+                        cache=cache,
+                        genesis_transactions=write_ledger_config.get(
+                            "genesis_transactions"
+                        ),
+                        read_only=write_ledger_config.get("read_only"),
+                        socks_proxy=write_ledger_config.get("socks_proxy"),
+                    ),
+                    ref(self),
+                ),
+            )
+            if (
+                "endorser_alias" in write_ledger_config
+                and "endorser_did" in write_ledger_config
+            ):
+                self.settings["endorser.endorser_alias"] = write_ledger_config.get(
+                    "endorser_alias"
+                )
+                self.settings["endorser.endorser_public_did"] = write_ledger_config.get(
+                    "endorser_did"
+                )
+        elif self.ledger_pool:
```

Comment:
I know Indy SDK wallet is on its way out the door but this is an almost exact copy-paste of what's in the askar profile. Is it possible to reuse at least some of the logic so if issues are discovered later we don't have to change it in both places?